lib/kafka/sasl_plain_authenticator.rb

```diff
@@ -0,0 +1,37 @@
+module Kafka
+  class SaslPlainAuthenticator
```

Comment:
Agreed.  I broke the `FakeServer` class out and added support for SASL PLAIN authentication.